src/mscorlib/src/System/Activator.cs

```diff
@@ -2,140 +2,77 @@
-////////////////////////////////////////////////////////////////////////////////
-//
-// Activator is an object that contains the Activation (CreateInstance/New) 
-//  methods for late bound support.
-//
-// 
-// 
-//
+using System.Reflection;
+using System.Globalization;
 
 namespace System
 {
-    using System;
-    using System.Reflection;
-    using System.Security;
-    using CultureInfo = System.Globalization.CultureInfo;
-    using StackCrawlMark = System.Threading.StackCrawlMark;
-    using System.Runtime.InteropServices;
-    using System.Runtime.CompilerServices;
-    using AssemblyHashAlgorithm = System.Configuration.Assemblies.AssemblyHashAlgorithm;
-    using System.Runtime.Versioning;
-
-    // Only statics, does not need to be marked with the serializable attribute
-    public sealed class Activator
+    /// <summary>
+    /// Activator contains the Activation (CreateInstance/New) methods for late bound support.
+    /// </summary>
+    public static class Activator
     {
-        internal const int LookupMask = 0x000000FF;
-        internal const BindingFlags ConLookup = (BindingFlags)(BindingFlags.Instance | BindingFlags.Public);
         internal const BindingFlags ConstructorDefault = BindingFlags.Instance | BindingFlags.Public | BindingFlags.CreateInstance;
 
-        // This class only contains statics, so hide the worthless constructor
-        private Activator()
-        {
-        }
-
-        // CreateInstance
-        // The following methods will create a new instance of an Object
-        // Full Binding Support
-        // For all of these methods we need to get the underlying RuntimeType and
-        //  call the Impl version.
-        static public Object CreateInstance(Type type,
-                                            BindingFlags bindingAttr,
-                                            Binder binder,
-                                            Object[] args,
-                                            CultureInfo culture)
-        {
-            return CreateInstance(type, bindingAttr, binder, args, culture, null);
-        }
+        public static object CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, object[] args, CultureInfo culture) =>
+            CreateInstance(type, bindingAttr, binder, args, culture, null);
 
-        static public Object CreateInstance(Type type,
-                                            BindingFlags bindingAttr,
-                                            Binder binder,
-                                            Object[] args,
-                                            CultureInfo culture,
-                                            Object[] activationAttributes)
+        public static object CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, object[] args, CultureInfo culture, object[] activationAttributes)
         {
-            if ((object)type == null)
+            if (type == null)
```

Comment:
I think `(object)` cast was intentional here. It will call overloaded `Type` == operator otherwise that is quite a bit slower than a simple null check.

Reply:
You can use the new shiny C# `if (type is null)`. It will avoid the overloaded operator as well.

Reply:
Ah, for some reason I was under the impression that newer C# compilers would emit `ceq` when comparing to null. Will fix.

Reply:
I was thinking of strings. Checking a string for null will not call string's == operator.